live-build/misc/upgrade-scripts/verify-impl

```diff
@@ -26,6 +26,25 @@ function usage() {
 	exit 2
 }
 
+function get_container_name() {
```

Comment:
I'd be OK if it didn't work with `machinectl shell` since we can use `upgrade-container run` to execute commands in the container. If we wanted to, we could add a `upgrade-container run-interactive` (maybe with a better name) to get an interactive shell in the container, since `run` is meant to be run non-interactively.

This way, we could update the `upgrade-container` script to expose the environment variable, and we'd "hide" the technology (`systemd-nspawn` and/or `machinectl`) that's used to create the container from users (both from actual people, but also from scripts and other software using the container).

With that said, if we want to expose the fact that we're using `systemd-nspawn` for the container, and encourage folks to use things like `machinectl` and `systemd-run` directly, I'm OK with that too. It just means that we're more closely tied to `systemd-nspawn` and the tools it provides, and changing that would be much more difficult; forcing folks to use `upgrade-container run` gives us a layer of abstraction, so we can change the underlying tools without changing the users of the script.

So, I just wanted to bring up an alternative solution, in case it wasn't considered; ultimately, I'm OK going this route if we deem this best (i.e. encouraging the use of tools like `machinectl`, which implies we're exposing the fact that we're using a `systemd-nspawn` container to users).